### ISSUE TITLE
Create addon and MidiManager in C#

### ISSUE THREAD

**Issue by nlaha** [nlaha/godot-midi-4.0#7]
Currently, all the core logic should be available both to GDScript and C#, but the actual addon is written in GDScript.

In other words, the contents of https://github.com/nlaha/godot-midi-4.0/tree/main/game/addons/godot_midi

Should be written in C# alongside the GDScript implementation.

**Comment by aidan-j-rhoden** [comment]
I might take a shot at this; just to be clear, the contents of `/addons/godot_midi/` need to be recreated in C#?

**Comment by aidan-j-rhoden** [comment]
Also, should this be noted? (Godotengine.org - C# basics)[https://docs.godotengine.org/en/stable/tutorials/scripting/c_sharp/c_sharp_basics.html#current-gotchas-and-known-issues:~:text=Writing%20editor%20plugins%20is%20possible%2C%20but%20it%20is%20currently%20quite%20convoluted.]

**Comment by nlaha** [comment]
I've started working on a rewrite of how the MIDI parser interfaces with Godot. The new version no longer relies on the animation system. For that reason, I suggest you wait until I have that in a working state, as much of the Godot addon logic will be changed to accommodate the new system.

I'm closing this issue for now, but will re-open it once the new version (written in GDScript) is merged into main.

**Comment by aidan-j-rhoden** [comment]
Let me note real quick, ChatGPT worked phenomenally for this task.  I did one script myself, and after debugging for an hour or so, got it working.  I gave GPT the script and this prompt: `Translate the following code from GDScript to C#:`

It worked flawlessly, and in ~2 seconds.

**Comment by nlaha** [comment]
> Let me note real quick, ChatGPT worked phenomenally for this task. I did one script myself, and after debugging for an hour or so, got it working. I gave ChatGPT the script and this prompt: `Translate the following code from GDScript to C#:`
> 
> It produced the exact same working result I created, and in ~2 seconds.

I'm going to keep this issue closed as I was able to write a MIDI playback node ("MidiPlayer") in C++, thus no GDScript or C# version is required.
